cf-agent/package_module.c

```diff
@@ -742,14 +742,17 @@ int UpdatePackagesDB(Rlist *data, const char *pm_name, UpdateType type)
         char *inventory_list = BufferClose(inventory_data);
 
         /* We can have empty list of installed software or available updates. */
-        if (!inventory_list)
+        if (inventory_list == NULL || inventory_list[0] == '\0')
```

Comment:
@vpodzime I do find `\n` as a value a bit "mysterious" so would seem like the "best way" would be to do something like

```
if (inventory_list == NULL)
{
    WriteDB(db_cached, inventory_key, "", 0);
}
```

so that regardless of why `inventory_list` is `NULL` or "empty" we get the same reasonable result not mysterious `\n`. If `\n` is some magic lmdb style "empty indicator" we should say so here.

and adjust `cf-check` to not `assert (size != 0)`.